geoapps/inversion/components/topography.py

```diff
@@ -96,15 +97,30 @@ def active_cells(self, mesh: InversionMesh, data: InversionData) -> np.ndarray:
         :return: active_cells: Mask that restricts a model to the set of
             earth cells that are active in the inversion (beneath topography).
         """
-        if self.params.inversion_type == "magnetotellurics":
+        forced_to_surface = [
+            "magnetotellurics",
+            "direct current 3d",
+            "induced polarization 3d",
+        ]
+        if self.params.inversion_type in forced_to_surface:
             active_cells = active_from_xyz(
                 mesh.mesh, self.locations, grid_reference="bottom_nodes", logical="any"
             )
+            print(
+                "Adjusting active cells so that receivers are all within an active cell . . ."
+            )
+
             active_cells[
                 mesh.mesh._get_containing_cell_indexes(  # pylint: disable=protected-access
                     data.locations
                 )
             ] = True
+
+            if floating_active(mesh.mesh, active_cells):
+                warnings.warn(
+                    "Active cell adjustment has created a patch of active cells in the air, likely due to a faulty survey location."
```

Comment:
Should probably unittest this part

Reply:
I wrote unit tests for the floating_active and face_average methods